File: USACO/Problems/2018 DEC Bronze #1 - Mixing Milk/18_Dec_Bronze_1_mixingMilk.py
#파일 실행시 반드시 폴더 맨 앞으로 보내야함. CS 폴더로 보내기. 아니면 오류 남

totalOperation = 100
numBucket = 3

capacity = [0 for _ in range(numBucket)]
amount = [0 for _ in range(numBucket)]

mixmilkIn = open('mixmilk.in', 'r', encoding='utf-8')
for i in range(numBucket):
    capacity[i], amount[i] = map(int, mixmilkIn.readline().split())

for i in range(totalOperation):
    bucket1 = i % 3
    bucket2 = (i + 1) % 3
    amountPour = min(capacity[bucket2] - amount[bucket2], amount[bucket1])
    amount[bucket1] -= amountPour
    amount[bucket2] += amountPour

mixmilkOut = open('mixmilk.out', 'w', encoding='utf-8')
for m in amount:
    mixmilkOut.write(str(m) + "\n")

File: USACO/Problems/2018 DEC Bronze #1 - Mixing Milk/18_Dec_Bronze_1_mixingMilk.cpp
#include <iostream>
#include <fstream> // 파일 입출력을 위한 헤더 파일
#include <string>  // 문자열을 사용하기 위한 헤더 파일
#include <vector>  // 벡터(동적 배열)를 사용하기 위한 헤더 파일
#include <algorithm> // min, max 함수를 사용하기 위한 헤더 파일

using namespace std;

const int numBucket = 3;
const int totalOperation = 100;

int main(){
    freopen("mixmilk.in", "r", stdin);

    vector<int> amount(numBucket);
    vector<int> capacity(numBucket);

    for (int i = 0; i < numBucket; i++) {
        scanf("%d %d", &capacity[i], &amount[i]);
    }

    for (int i = 0; i < totalOperation; i++) {
        int bucket1 = i % numBucket;
        int bucket2 = (i + 1) % numBucket;

        int amountPour = min(amount[bucket1], capacity[bucket2] - amount[bucket2]);
        amount[bucket1] -= amountPour;
        amount[bucket2] += amountPour;
    }

    freopen("mixmilk.out", "w", stdout);
    for (int i = 0; i < numBucket; i++) {
        printf("%d\n", amount[i]);
    }
}